### PR TITLE
migrate vesting contracts

### PR DESCRIPTION
**Pull request by mintdart**

### CODE REVIEW

File: components/Vesting/Table/CustomValues/Migrate.tsx [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+  const checkingApproval = fetchingTokenApproval || fetchingTokenApproval1;
+  const { mutate: approveTokenSpend, isLoading: approvingToken, error: errorApproving } = useApproveToken();
+  const { chainId } = useNetworkProvider();
+  function approveToken() {
+    if (chainId) {
+      approveTokenSpend(
+        {
+          tokenAddress: getAddress(data.token),
+          amountToApprove: toVest.toFixed(),
+          // TODO update it to networkDetails[chainId].vestingFactory
```
Comment (0xngmi) [inline components/Vesting/Table/CustomValues/Migrate.tsx:96]: is this todo wrong?

**Review by 0xngmi — CHANGES_REQUESTED** [review]

File: components/Stream/Table/CustomValues/TotalStreamed.tsx [hunk 2]
```diff
@@ -3,20 +3,24 @@ import { useIntl } from 'next-intl';
...
-  const [amount, setAmount] = React.useState<string | null>(null);
+  const ref = React.useRef<HTMLParagraphElement>(null);
 
   const intl = useIntl();
 
   React.useEffect(() => {
     const id = setInterval(() => {
-      setAmount(intl.formatNumber(totalStreamedFormatter(data), { maximumFractionDigits: 5 }));
+      if (ref.current) {
+        if (ref.current) {
```
[the remark this answers is not in the document]
Comment (mintdart) [reply components/Stream/Table/CustomValues/TotalStreamed.tsx]: it shouldn't be, removed it

File: components/Vesting/create/index.tsx [hunk 3]
```diff
@@ -104,7 +106,6 @@ export default function CreateVesting({ factory }: { factory: string }) {
 
     const isEOA = (await provider.getCode(recipientAddress)) === '0x' ? true : false;
     if (!isEOA) {
-      setRecipient(recipientAddress);
```
[the remark this answers is not in the document]
Comment (mintdart) [reply components/Vesting/create/index.tsx:107]: setting it here in handleChange https://github.com/LlamaPay/interface/pull/104/files/4dd9166dc3f1f55c428b940b3ed025aa68b597c2#diff-7f608adae12abf7a4ebeecbe2dbaef68c48b28136ce07859d06790466c7ae5a2R168 , if we set recipient inside onSubmit , can't show `Recipient can't be same as the funder warning` here https://github.com/LlamaPay/interface/pull/104/files/4dd9166dc3f1f55c428b940b3ed025aa68b597c2#diff-7f608adae12abf7a4ebeecbe2dbaef68c48b28136ce07859d06790466c7ae5a2R171

File: components/Vesting/Table/CustomValues/Migrate.tsx (hunk 1 above)
Comment (mintdart) [reply components/Vesting/Table/CustomValues/Migrate.tsx]: yes, removed it

**Pull request merged**